tests/benchmark.json

```diff
@@ -1,15 +1,15 @@
 {
   "time": {
-    "initialize": 0.007,
-    "run": 3.792,
-    "postprocess": 0.129,
-    "total": 3.927
+    "initialize": 0.006,
+    "run": 4.548,
+    "postprocess": 0.227,
+    "total": 4.782
```

Comment:
I think this is because my computer is slow?! When I run update_baseline on the `optimise-performance` branch, i get `"total": 4.049`, so this represents a speed-up.

Running example_sim.py takes 8.5s on this branch for me, compared to 12.4s on the `optimise-performance` branch. So depending on the sim I think the performance gains could be significant. Keen to see what you find @pausz. I found your benchmark.py script incredibly useful for this!

Reply:
This is great @robynstuart!  

I hadn't touched benchmark.json because things were running slower on my computer (also close to 5-ish s). btw, Cliff wrote the benchmark script :), I just updated it a bit. 
 
#### Runtime comparison benchmark.py, assessing `sim.run()` : 

On main (with changes from `optimise-performance`)
Average runtime (over 20 runs): 6.2778 s

On this branch 
Average runtime (over 20 runs): 5.3839 s
```diff
+15% speed-up
```

#### Runtime comparison example_sim.py : 
On main (with changes from `optimise-performance`)
Average runtime (over 20 runs): 4.188 s


On this branch 
Average runtime (over 20 runs): 3.5835 s
```diff
+15% speed-up
```

Reply:
@MObrien-IDM ready for you to review!